src/base_classes/baseball.py

```diff
@@ -0,0 +1,175 @@
+                return "Upcoming"
+            else:
+                return status
+                
+        except Exception as e:
+            self.logger.error(f"Error getting baseball game status: {e}")
+            return ""
+
+
+class BaseballLive(Baseball):
+    """Base class for live baseball games."""
+    
+    def __init__(self, config: Dict[str, Any], display_manager, cache_manager, logger: logging.Logger, sport_key: str):
+        super().__init__(config, display_manager, cache_manager, logger, sport_key)
+        self.logger.info(f"{sport_key.upper()} Live Manager initialized")
+    
+    def _should_show_baseball_game(self, game: Dict) -> bool:
+        """Determine if a baseball game should be shown."""
+        try:
+            # Only show live games
+            if not self._is_baseball_game_live(game):
+                return False
+            
+            # Check if game meets display criteria
+            return self._should_show_game(game)
+            
+        except Exception as e:
+            self.logger.error(f"Error checking if baseball game should be shown: {e}")
+            return False
+
+
+class BaseballRecent(Baseball):
+    """Base class for recent baseball games."""
+    
+    def __init__(self, config: Dict[str, Any], display_manager, cache_manager, logger: logging.Logger, sport_key: str):
+        super().__init__(config, display_manager, cache_manager, logger, sport_key)
+        self.logger.info(f"{sport_key.upper()} Recent Manager initialized")
+    
+    def _should_show_baseball_game(self, game: Dict) -> bool:
+        """Determine if a recent baseball game should be shown."""
+        try:
+            # Only show final games
+            if not game.get('is_final', False):
+                return False
+            
+            # Check if game meets display criteria
+            return self._should_show_game(game)
+            
+        except Exception as e:
+            self.logger.error(f"Error checking if baseball game should be shown: {e}")
+            return False
+
+
+class BaseballUpcoming(Baseball):
+    """Base class for upcoming baseball games."""
+    
+    def __init__(self, config: Dict[str, Any], display_manager, cache_manager, logger: logging.Logger, sport_key: str):
+        super().__init__(config, display_manager, cache_manager, logger, sport_key)
+        self.logger.info(f"{sport_key.upper()} Upcoming Manager initialized")
+    
+    def _should_show_baseball_game(self, game: Dict) -> bool:
+        """Determine if an upcoming baseball game should be shown."""
+        try:
+            # Only show upcoming games
+            if not game.get('is_upcoming', False):
+                return False
+            
+            # Check if game meets display criteria
+            return self._should_show_game(game)
+            
+        except Exception as e:
+            self.logger.error(f"Error checking if baseball game should be shown: {e}")
+            return False
```

Comment:
THis really shouldn't be needed. The Individual Leagues just extend the common `SportsRecent` and `SportsUpcoming` since they all display the same data. The only reason to make a specific sport recent/upcoming would be to extend the update/display method of the class to show different data for all leagues of a sport.